src/data/sigma/basic.lean

```diff
@@ -148,6 +148,12 @@ by cases x; refl
 lemma prod.snd_to_sigma {α β} (x : α × β) : (prod.to_sigma x).snd = x.snd :=
 by cases x; refl
 
+-- we generate this manually as `@[derive has_reflect]` fails
+meta instance sigma.reflect {α : Type} (β : α → Type)
```

Comment:
We're not in the `sigma` namespace here, only a section.

Note that `vector.reflect` is also not in core; I think it's not worth pushing too many things into core with Lean4 on the horizon, especially since I will need to modify this again after #14766